pkg/apis/core/v1beta1/types.go

```diff
@@ -362,6 +380,127 @@ type StorageProfileList struct {
 	Items []StorageProfile `json:"items"`
 }
 
+// DataSource references an import/clone source for a DataVolume
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type DataSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataSourceSpec   `json:"spec"`
+	Status DataSourceStatus `json:"status,omitempty"`
+}
+
+// DataSourceSpec defines specification for DataSource
+type DataSourceSpec struct {
+	// Source is the source of the data referenced by the DataSource
+	Source DataSourceSource `json:"source"`
+}
+
+// DataSourceSource represents the source for our DataSource
+type DataSourceSource struct {
+	// +optional
+	PVC *DataVolumeSourcePVC `json:"pvc,omitempty"`
+}
+
+// DataSourceStatus provides the most recently observed status of the DataSource
+type DataSourceStatus struct {
+	Conditions []DataSourceCondition `json:"conditions,omitempty" optional:"true"`
+}
+
+// DataSourceCondition represents the state of a data source condition
+// FIXME: needs LastTransitionTime/LastHeartbeatTime/Reason/Message?
+type DataSourceCondition struct {
+	Type   DataSourceConditionType `json:"type" description:"type of condition ie. Ready"`
+	Status corev1.ConditionStatus  `json:"status" description:"status of the condition, one of True, False, Unknown"`
+}
+
+// DataSourceConditionType is the string representation of known condition types
+type DataSourceConditionType string
+
+// DataSourceList provides the needed parameters to do request a list of Data Sources from the system
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataSourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items provides a list of DataSources
+	Items []DataSource `json:"items"`
+}
+
+// DataImportCron defines a cron job for recurring polling/importing disk images as PVCs into a golden image namespace
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type DataImportCron struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataImportCronSpec   `json:"spec"`
+	Status DataImportCronStatus `json:"status,omitempty"`
```

Comment:
From what I think to remember about other resources, status usually is {} in the yaml if empty.
So it might not be omitempty usually.

Reply:
Status of DataVolume and StorageProfile is `omitempty` and no pointer. What's different here?

Reply:
Yeah I think this is fine. Seen this a few places now and seems normal.

Reply:
`omitempty` really makes no sense with non-pointer struct members.  Either way it will get serialized as `{}`

Reply:
I understand it is used in a lot of places where it doesn't make sense.  Just being pedantic.

Reply:
Ok, so just removing the `omitempty`.

Reply:
ah, forgot that that tag is used for schema generation and without "omitempty" it will be required, so I guess add it back.  I did not consider the schema impact.

Reply:
re-added `omitempty` :)